### ISSUE TITLE
Quick terminal splits get 1px out of sync with aggressive resizing

### ISSUE THREAD

**Issue by JakeWharton** [ghostty-org/ghostty#2944]
I'm on macOS 14.7.1 (23H222) with Ghostty 9f75d93a.

My full config is
```
initial-window = false
copy-on-select = true
window-save-state = always
background-opacity = 0.9
background-blur-radius = 5
window-step-resize = true
keybind = global:alt+grave_accent=toggle_quick_terminal
mouse-hide-while-typing = true
macos-option-as-alt = true
```

I tend to keep the quick terminal split into two panes. Aggressive resizing causes them to get 1px out of sync in terms of height:

![Image](https://github.com/user-attachments/assets/8a64cd8e-437d-4c2c-8e7c-e3fa3f45b19a)

https://github.com/user-attachments/assets/4889521c-acc0-432f-b4ed-330c1a332276

Hiding and re-showing the quick terminal fixes this misalignment.

This issue occurs regardless of the `window-step-resize = true` settings.

**Comment by blychs** [comment]
I just attempted to reproduce this in MacOS (14.6.1) and the default config and could not. It looks as expected, without the weird mismatch

**Comment by JakeWharton** [comment]
With 1.0.0 release, I'm unfortunately still able to repro:

![Image](https://github.com/user-attachments/assets/4d8ba6d6-d86f-439c-8890-937b59d2aecb)

**Comment by mitchellh** [comment]
Thanks. I’ve done nothing to try to solve this yet. I haven’t yet been able to reproduce it.

**Comment by JakeWharton** [comment]
Yeah not a big deal. It's _extremely_ low priority, and hopefully not even your problem but something in SwiftUI/AppKit or wherever. Mostly just confirming that I _can_ still repro it with wildly-aggressive resizing (which is not at all like the resizing that I expect 99.99% of people to do). I don't even do such resizing normally–it was only when I was testing my TUI frameworks parsing of in-band resizing notifications that caused me to discover it…

**Comment by albinkc** [comment]
had this with zero config [video is normal speed]. Funnily enough, could not replicate when resizing from the left side.

Ghostty 1.0.1

https://github.com/user-attachments/assets/8d236dd0-a75f-4ec9-a63b-0e41a0159046

**Comment by albinkc** [comment]
Here's another one.

```config
shell-integration = zsh
cursor-click-to-move = true
```
https://github.com/user-attachments/assets/ee94275e-7afc-46a4-8f5a-9d2bdd515c36

**Comment by mitchellh** [comment]
I'm pretty sure this is fixed along the way. We've mostly rewritten our quick terminal since this was reported and I haven't seen any other reports. If there are, please open a new discussion!
